parser_library/src/parsing/grammar/ca_expr_rules.g4

```diff
@@ -303,19 +328,16 @@ ca_dupl_factor returns [ca_expr_ptr value]
 	|;
 
 substring returns [expressions::ca_string::substring_t value]
-	: lpar e1=expr comma e2=expr rpar
+	: lpar e1=expr comma (e2=expr|ASTERISK) rpar
 	{
 		$value.start = std::move($e1.ca_expr);
-		$value.count = std::move($e2.ca_expr);
-		$value.substring_range = provider.get_range($lpar.ctx->getStart(), $rpar.ctx->getStop());
-		resolve_expression($value.start, context::SET_t_enum::A_TYPE);
-		resolve_expression($value.count, context::SET_t_enum::A_TYPE);
-	}
-	| lpar expr comma ASTERISK rpar
-	{
-		$value.start = std::move($expr.ca_expr);
 		$value.substring_range = provider.get_range($lpar.ctx->getStart(), $rpar.ctx->getStop());
 		resolve_expression($value.start, context::SET_t_enum::A_TYPE);
+		if ($e2.ctx)
```

Comment:
Did you mean `$e2.ca_expr`?

Reply:
The `$e2.ctx` tests the presence of the `expr` parsing context (aliased by `e2` label). This tests which path (expression or asterisk) the parsed had chosen. The `$e2.ca_expr` does not exist unless `$e2.ctx != nullptr`.

Reply:
👍